src/Symfony/Component/HttpKernel/Fragment/FragmentHandler.php

```diff
@@ -97,7 +97,8 @@ public function render($uri, string $renderer = 'inline', array $options = [])
     protected function deliver(Response $response)
     {
         if (!$response->isSuccessful()) {
-            throw new \RuntimeException(sprintf('Error when rendering "%s" (Status code is %d).', $this->requestStack->getCurrentRequest()->getUri(), $response->getStatusCode()));
+            $responseStatusCode = $response->getStatusCode();
+            throw new \RuntimeException(sprintf('Error when rendering "%s" (Status code is %d).', $this->requestStack->getCurrentRequest()->getUri(), $responseStatusCode), $responseStatusCode);
```

Comment:
I would use a `HttpException` here instead so that the error code is understood as a status code, seem more semantic to me (and as it extends `RuntimeException`, no BC break).